x-pack/platform/plugins/shared/stack_connectors/server/connector_types/openai/openai.ts

```diff
@@ -76,24 +81,73 @@ export class OpenAIConnector extends SubActionConnector<Config, Secrets> {
       ...('projectId' in this.config ? { 'OpenAI-Project': this.config.projectId } : {}),
     };
 
-    this.openAI =
-      this.config.apiProvider === OpenAiProviderType.AzureAi
-        ? new OpenAI({
-            apiKey: this.secrets.apiKey,
-            baseURL: this.config.apiUrl,
-            defaultQuery: { 'api-version': getAzureApiVersionParameter(this.config.apiUrl) },
-            defaultHeaders: {
-              ...this.headers,
-              'api-key': this.secrets.apiKey,
-            },
-          })
-        : new OpenAI({
-            baseURL: removeEndpointFromUrl(this.config.apiUrl),
-            apiKey: this.secrets.apiKey,
-            defaultHeaders: {
-              ...this.headers,
-            },
-          });
+    try {
+      if (
+        this.provider === OpenAiProviderType.Other &&
+        ('certificateFile' in this.config ||
```

Comment:
I reverted some of your changes such as `configAny` to keep the code clean. However, I need you to make some fixes to the methods in this file (see comments)